### DISCUSSION TITLE
Single file download / permalink

### DISCUSSION THREAD

**Discussion by maartenbreddels** [esm-dev/esm.sh#778] [Ideas]
Hi,

If I look at https://esm.sh/*antd@5.12.4 (note the * to get rid of dependencies) I see that it only imports a single file as expected. Is there a way to get a direct link to this single file, or get a redirect to it? I'd like to automatically fetch this file (for offline support).

Regards,

Maarten

**Comment by ije** [comment]
do you mean this link https://esm.sh/v135/antd@5.12.4/X-ZS8q/es2022/antd.mjs ?

**Reply by maartenbreddels** [reply]
Yes, is there a way to get only the content of that? Without having to do some scripting.

**Reply by ije** [reply]
there is a `x-esm-id` header points to the link

**Reply by mb21** [reply]
Came here wondering the same. Is this `x-esm-id` header documented somewhere?

Is this replacing the `X-Esm-Deps` header? From [v127 release nots](https://github.com/esm-dev/esm.sh/releases/tag/v127):

> For framework authors, esm.sh provides an extra X-Esm-Deps header that allows you to get those preload import urls.

Why is this only recommended for framework authors? Seems like avoiding an extra http request is always beneficial?

**Reply by ije** [reply]
it has been removed, module has heavy deps may send a huge header that reach cf's limlit

**Reply by mb21** [reply]
but `x-esm-id` is still supported? is it documented anywhere?

**Reply by ije** [reply]
yes, however it has been renamed to `x-esm-path`
